src/curve.rs

```diff
@@ -231,6 +231,9 @@ pub trait Identity {
     /// Returns the identity element of the curve.
     /// Can be used as a constructor.
     fn identity() -> Self;
+
```

Comment:
I'm not sure whether this should be in the same trait, because I don't see why everything that implements identity() in order to have a constructor now has to have equality testing...